pkg/controller/constraint/constraint_controller.go

```diff
@@ -620,78 +619,6 @@ func (c *ConstraintsCache) reportTotalConstraints(ctx context.Context, reporter
 	}
 }
 
-func IsVapAPIEnabled() (bool, *schema.GroupVersion) {
-	vapMux.RLock()
-	if VapAPIEnabled != nil {
-		apiEnabled, gvk := *VapAPIEnabled, GroupVersion
-		vapMux.RUnlock()
-		return apiEnabled, gvk
-	}
-
-	vapMux.RUnlock()
-	vapMux.Lock()
-	defer vapMux.Unlock()
-
-	if VapAPIEnabled != nil {
-		return *VapAPIEnabled, GroupVersion
-	}
-	config, err := rest.InClusterConfig()
-	if err != nil {
-		log.Info("IsVapAPIEnabled InClusterConfig", "error", err)
-		VapAPIEnabled = new(bool)
-		*VapAPIEnabled = false
-		return false, nil
-	}
-	clientset, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		log.Info("IsVapAPIEnabled NewForConfig", "error", err)
-		*VapAPIEnabled = false
-		return false, nil
-	}
-
-	groupVersion := admissionregistrationv1.SchemeGroupVersion
-	resList, err := clientset.Discovery().ServerResourcesForGroupVersion(groupVersion.String())
-	if err == nil {
-		for i := 0; i < len(resList.APIResources); i++ {
-			if resList.APIResources[i].Name == "validatingadmissionpolicies" {
-				VapAPIEnabled = new(bool)
-				*VapAPIEnabled = true
-				GroupVersion = &groupVersion
-				return true, GroupVersion
-			}
-		}
-	}
-
-	groupVersion = admissionregistrationv1beta1.SchemeGroupVersion
-	resList, err = clientset.Discovery().ServerResourcesForGroupVersion(groupVersion.String())
-	if err == nil {
-		for i := 0; i < len(resList.APIResources); i++ {
-			if resList.APIResources[i].Name == "validatingadmissionpolicies" {
-				VapAPIEnabled = new(bool)
-				*VapAPIEnabled = true
-				GroupVersion = &groupVersion
-				return true, GroupVersion
-			}
-		}
-	}
-
-	log.Error(err, "error checking VAP API availability", "IsVapAPIEnabled", "false")
-	VapAPIEnabled = new(bool)
-	*VapAPIEnabled = false
-	return false, nil
-}
-
-func vapBindingForVersion(gvk schema.GroupVersion) (client.Object, error) {
```

Comment:
I agree with @maxsmythe on 

> Actually, reflecting on moving vapForVersion and vapBindingForVersion, I think those should stay with the controller code.

Let's not move there two methods.

Reply:
Done